eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/BackCompatTest.java

```diff
@@ -0,0 +1,132 @@
+    private EventHubProducer sender;
+    private EventHubConsumer receiver;
+    private ReactorHandlerProvider handlerProvider;
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @Override
+    protected String testName() {
+        return testName.getMethodName();
+    }
+
+    @Override
+    protected void beforeTest() {
+        logger.asInfo().log("[{}]: Performing test set-up.", testName.getMethodName());
+
+        handlerProvider = new ReactorHandlerProvider(getReactorProvider());
+        client = new EventHubClient(getConnectionOptions(), getReactorProvider(), handlerProvider);
+    }
+
+    @Override
+    protected void afterTest() {
+        logger.asInfo().log("[{}]: Performing test clean-up.", testName.getMethodName());
+
+//        if (client != null) {
+//            client.close();
+//        }
+//
+//        if (sender != null) {
+//            try {
+//                sender.close();
+//            } catch (IOException e) {
+//                logger.asError().log(String.format("[%s]: Sender doesn't close properly.", testName.getMethodName()), e);
+//            }
+//        }
+//
+//        if (receiver != null) {
+//            try {
+//                receiver.close();
+//            } catch (IOException e) {
+//                logger.asError().log(String.format("[%s]: Receiver doesn't close properly.", testName.getMethodName()), e);
+//            }
+//        }
+    }
+
+    /**
+     * Verifies test work with versions before 0.11.0.
+     */
+    @Ignore
+    @Test
+    public void backCompatWithJavaSDKOlderThan0110() {
+        skipIfNotRecordMode();
+
+        final Message originalMessage = Proton.message();
+        // Arrange
+        receiver = client.createConsumer(getConsumerGroupName(), PARTITION_ID, EventPosition.latest(),
+            new EventHubConsumerOptions());
+        // until version 0.10.0 - we used to have Properties as HashMap<String,String>
+        // This specific combination is intended to test the back compat - with the new Properties type as HashMap<String, Object>
+        final HashMap<String, Object> appProperties = new HashMap<>();
+        appProperties.put(APPLICATION_PROPERTY, "value1");
+        appProperties.put(INT_APPLICATION_PROPERTY, "3");
+        // back compat end
+        final ApplicationProperties applicationProperties = new ApplicationProperties(appProperties);
+
+        originalMessage.setApplicationProperties(applicationProperties);
+        originalMessage.setBody(new Data(Binary.create(ByteBuffer.wrap(PAYLOAD.getBytes()))));
+        originalMessage.setMessageAnnotations(new MessageAnnotations(new HashMap<>()));
+        final EventData msgEvent = new EventData(originalMessage);
+
+
+        // Act & Assert
+        Flux<EventData> receivedEventData = receiver.receive();
+        sender = client.createProducer(new EventHubProducerOptions().partitionId(PARTITION_ID).retry(Retry.getNoRetry()).timeout(Duration.ofSeconds(30)));
```

Comment:
This step is part of the arrange, because you're creating the sender here. If you keep the creation here, it should be in the block above.